src/java.desktop/share/classes/java/awt/Robot.java

```diff
@@ -164,30 +165,35 @@ public Robot(GraphicsDevice screen) throws AWTException {
     private void init(GraphicsDevice screen) throws AWTException {
         Toolkit toolkit = Toolkit.getDefaultToolkit();
         if (toolkit instanceof ComponentFactory) {
-            peer = ((ComponentFactory)toolkit).createRobot(screen);
+            peer = ((ComponentFactory) toolkit).createRobot(screen);
         }
         initLegalButtonMask();
+        GraphicsEnvironment ge = GraphicsEnvironment.getLocalGraphicsEnvironment();
+        GraphicsDevice[] gs = ge.getScreenDevices();
+        for (int i = 0; i < gs.length; i++) {
+            allScreenBounds[i] = gs[i].getDefaultConfiguration().getBounds();
+        }
```

Comment:
This doesn't look right to me… `init` is an instance method, yet it initialises `allScreenBounds` which is shared among all instances.

In a way it makes sense to share the graphics environment / configuration. On the other hand how much would it affect the operation of robot if the bounds are fetched from `GraphicsEnvironment` only when needed?

What happens if the graphics configuration changes? Does the user need to create a new instance of the `Robot` class?

Reply:
I did consider this and thought it would be too expensive to fetch the bounds on every mouseMove action, so i thought it would be better to reinitialize robot when the graphics configuration changes. Do you have a better idea for where/when to cache the bounds?

Reply:
The first question to answer: _Is it expensive to get screen data from GraphicsEnvironment?_ It shouldn't be, the data is already there, you just fetch the data.

> it would be better to reinitialize robot when the graphics configuration changes

But I see no code which re-initialises the cached graphics configuration if anything changes.

Do I miss anything?

Reply:
> But I see no code which re-initialises the cached graphics configuration if anything changes.
> 
> Do I miss anything?

What I meant was have the user reinitialize the instance of robot, but it's not a clean solution. I've now updated it to grab the screen data during the mouseMove.